packages/portal-proto/src/components/Modals/SetModals/SaveSelectionModal.tsx

```diff
@@ -165,9 +166,12 @@ const SaveSelectionAsSetModal: React.FC<SaveSelectionAsSetModalProps> = ({
                   color: "red",
                   closeButtonProps: { "aria-label": "Close notification" },
                 });
-              })
+              });
+          }}
+          disabled={!form.isValid()}
+          leftSection={
+            response?.isLoading && <Loader size="sm" color="white" />
```

Comment:
For typing reasons, I think its generally best to avoid this way of rendering when passing in a component as a prop. When the condition is false, you're passing in the value of false instead of undefined to `leftSection`. Depending on the how the client library is written, it could be a type error or just behave in a way you're not expecting. Looks like mantine has this prop typed as `ReactNode` so a boolean is fine but if they choose `ReactElement` then false would be a type error.

Reply:
should I use ternary and pass in `null`?

Reply:
Ternary and undefined

Reply:
updated